src/ORM/Behavior/TranslateBehavior.php

```diff
@@ -272,6 +272,43 @@ public function beforeSave(Event $event, EntityInterface $entity, ArrayObject $o
         $newOptions = [$this->_translationTable->getAlias() => ['validate' => false]];
         $options['associated'] = $newOptions + $options['associated'];
 
+        // Check early if empty transaltions are present in the entity.
+        // If this is the case, unset them to prevent persistence.
+        // This only applies if $this->_config['allowEmptyTranslations'] is false
+        if ($this->_config['allowEmptyTranslations'] === false) {
+            $translations = (array)$entity->get('_translations');
+            if (!empty($translations)) {
+                foreach ($translations as $locale => &$translation) {
+                    $fields = $translation->extract($this->_config['fields'], false);
+                    foreach ($fields as $field => $value) {
+                        if (empty($value)) {
+                            unset($translation->{$field});
+                        }
+                    }
+
+                    // Workaround to check the remaining properties
+                    $arrayEntity = $entity->toArray();
+
+                     // If now, the current locale property is empty,
+                     // unset it completely.
+                    if (empty($arrayEntity['_translations'][$locale])) {
```

Comment:
I don't think so, as `extract ` returns sth like this:
```
[
	'title' => null
]
```
instead of `[]` when using `toArray()`